pkg/kotsadm/push_images.go

```diff
@@ -61,20 +69,35 @@ func ExtractAirgapImages(archive string, destDir string, progressWriter io.Write
 	defer gzipReader.Close()
 
 	tarReader := tar.NewReader(gzipReader)
+	isFileInDir := false // when excludeImages is true, we only extract top level files and don't go into dirs
 	for {
 		header, err := tarReader.Next()
 		if err == io.EOF {
 			break
 		}
 
+		if header.Name == "." {
+			continue
+		}
+
 		if err != nil {
 			return errors.Wrap(err, "failed to read tar header")
 		}
 
+		if excludeImages && header.Typeflag == tar.TypeDir {
```

Comment:
IIRC, `tar.TypeDir` will only indicate if it's a directory but not if the file is "in a directory". e.g.:

/path/to -> header.Typeflag will be tar.TypeDir
/path/to/file -> header.Typeflag might/will _not_ be tar.TypeDir

Reply:
right, that's why `isFileInDir` variable is used. technically we don't need to continue once we see a directory, but just in case that changes.